orangecontrib/text/twitter.py

```diff
@@ -183,44 +242,40 @@ def search_authors(self, authors, *, max_tweets=0, collecting=False):
         if not collecting:
             self.reset()
 
-        if max_tweets == 0:     # set to max allowed for progress
+        if max_tweets == 0:  # set to max allowed for progress
             max_tweets = 3200
 
         if not isinstance(authors, list):
             authors = [authors]
 
-        cursors = [tweepy.Cursor(self.api.user_timeline, screen_name=a,
-                                 tweet_mode='extended')
-                   for a in authors]
-        corpus, count = self.fetch(cursors, max_tweets, search_author=True)
-        self.append_history('Author', authors, None, None, count)
+        cursors = [
+            tweepy.Cursor(
+                self.api.user_timeline, screen_name=a, tweet_mode="extended"
+            )
+            for a in authors
+        ]
+        corpus, count = self.fetch(
+            cursors, max_tweets, search_author=True, callback=callback
+        )
+        self.append_history("Author", authors, None, None, count)
         return corpus
 
-    def fetch(self, cursors, max_tweets, search_author):
+    def fetch(self, cursors, max_tweets, search_author, callback):
         if not isinstance(cursors, list):
             cursors = [cursors]
 
         count = 0
-        try:
-            for i, cursor in enumerate(cursors):
-                for j, tweet in enumerate(cursor.items(max_tweets), start=1):
-                    if self.should_break():
-                        break
-                    if tweet.id not in self.container:
-                        count += 1
-                    self.container[tweet.id] = tweet
-                    if j % 20 == 0:
-                        self.on_progress(len(self.container),
-                                         (i*max_tweets + j)/
-                                         (len(cursors)*max_tweets))
-                if self.should_break():
-                    break
-        except tweepy.TweepError as e:
```

Comment:
So the error is now handled in callback?

Reply:
Yep, it is now handled in the widget - on_exception. When an exception happens it is caught by the mixin and handled in exception. In the old widget, error appearance was controlled by API which is not possible with concurrent mixin and is IMO not a good practice. Now widget itself control error appearance.